# Options and their Uses

### BUSI 722: Data-Driven Finance II
### Kerry Back, Rice University

### Options and Open Interest

- A financial option is a right to buy or sell  a financial security.
    
- The right trades separately from the (underlying) security and usually even on a different exchange.
    
- The rights are not (usually) issued by the companies who issue the underlying securities.  
  - Instead, the rights are created when someone buys one from someone else.  
  - Open interest is the number that exist at any time.

### Example

- When a contract is first opened for trading, open interest is zero.
- Suppose Andy buys a contract from Chloe, and Brooke buys a contract from David.
  - Longs = Andy and Brooke
  - Shorts = Chloe and David
  - Open interest = 2

- Suppose Andy then sells a contract to David.
  - Andy: long + short = no position
  - David: short + long = no position
  - Longs = Brooke
  - Shorts = Chloe
  - Open interest = 1

### Clearinghouse

- The long party has an option.  The short party has an obligation.
- After a trade is made, the option clearinghouse steps in the middle and becomes the counterparty to both sides.

### Hedging, speculation, and income

- You pay upfront to acquire an option.  
  - The amount you pay is called the option premium.  
  - It is not part of the contract but instead is determined in the market (like a stock price).
- You buy options to hedge or to speculate.  You sell options for income.
- Sellers of options need to have sufficient equity in their accounts (margin).  A buyer needs enough cash to pay the premium but no more (like buying a stock).

###  Calls, puts, and strikes

- A call option gives the holder the right to buy an asset at a pre-specified price.

- A put option gives the holder the right to sell an asset at a pre-specified price.

- The asset is called the underlying asset or just the underlying.

- The pre-specified price is called the exercise price or strike price.  

### American and European

- An option is valid for a specified period of time, the end of which is called its expiration date or maturity date.

- Most financial options can be exercised at any time the owner wishes, prior to maturity.  Such options are called American.
    
- There are some options that can only be exercised on the maturity date.  They are called European.  Both types are traded on both continents.

### Moneyness

- Borrowing language from horse racing, we say a call is
  - in the money if the underlying price is above the strike,
  - at the money if the underlying price equals the strike
  - out of the money if the underlying price is below the strike
- The reverse for puts
 
- Also, "deep in the money" and "deep out of the money"

### Value of a call at maturity

- At maturity, the value of a call is

$$\begin{cases} 0 & \text{if underlying < strike}\\
 \text{underlying} - \text{strike} & \text{if underlying > strike}
 \end{cases}
 $$

- Equivalently, the value of a call is 

$$\max(\text{underlying price}-\text{strike}, 0)$$

In [8]:
import numpy as np
import plotly.express as px

strike = 50
underlying = np.linspace(0, 100, 100)
call = np.maximum(underlying-strike, 0)

fig = px.line(x=underlying, y=call)
fig.update_traces(
    hovertemplate="Underlying = $%{x:.2f}<br>Call = $%{y:.2f}<extra></extra>"
)
fig.update_layout(
    xaxis_title="Underlying Price",
    yaxis_title="Call Option Value",
    xaxis_title_font = {"size":20},
    yaxis_title_font =  {"size":20},
    template="plotly_white"
)
fig.show()

### Value of a put at maturity

- At maturity, the value of a put is

$$\begin{cases} \text{strike} - \text{underlying} & \text{if underlying < strike}\\
 0 & \text{if underlying > strike}
 \end{cases}
 $$

- Equivalently, the value of a put is 

$$\max(\text{strike}-\text{underlying price}, 0)$$

In [9]:
strike = 50
underlying = np.linspace(0, 100, 100)
put = np.maximum(strike-underlying, 0)

fig = px.line(x=underlying, y=put)
fig.update_traces(
    hovertemplate="Underlying = $%{x:.2f}<br>Put = $%{y:.2f}<extra></extra>"
)
fig.update_layout(
    xaxis_title="Underlying Price",
    yaxis_title="Put Option Value",
    xaxis_title_font = {"size":20},
    yaxis_title_font =  {"size":20},
    template="plotly_white"
)
fig.show()

### Option Data from Yahoo

- You can get current trading prices of stock options from finance.yahoo.com.
- You can click around and find it or use the yfinance library.
- We'll look at CVX options.

In [10]:
import yfinance as yf
cvx = yf.Ticker("cvx")

- cvx.options is the set of traded maturities
- cvx.option_chain("some date") is an object containing call and put data
- cvx.option_chain("some date").calls is a dataframe of call info
- cvx.option_chain("some date").puts is a dataframe of put info

In [11]:
date = cvx.options[6]
cvx.option_chain(date).calls

contractSymbol             lastTradeDate  strike  lastPrice    bid  \
0   CVX240315C00075000 2024-01-05 20:13:49+00:00    75.0      75.70  74.95   
1   CVX240315C00080000 2023-11-22 15:29:20+00:00    80.0      63.85  71.95   
2   CVX240315C00085000 2023-11-21 20:59:02+00:00    85.0      60.35  65.95   
3   CVX240315C00090000 2023-12-26 18:31:02+00:00    90.0      63.65  60.10   
4   CVX240315C00095000 2023-12-26 18:31:02+00:00    95.0      58.70  55.25   
5   CVX240315C00100000 2023-12-26 17:50:05+00:00   100.0      53.80  50.25   
6   CVX240315C00105000 2023-08-01 19:00:05+00:00   105.0      57.50  57.15   
7   CVX240315C00110000 2023-12-22 15:15:35+00:00   110.0      42.75  40.35   
8   CVX240315C00115000 2023-12-07 15:49:28+00:00   115.0      29.91  35.35   
9   CVX240315C00120000 2024-01-03 20:21:15+00:00   120.0      34.00  30.55   
10  CVX240315C00125000 2023-12-13 15:22:41+00:00   125.0      19.31  25.25   
11  CVX240315C00130000 2024-01-04 17:12:59+00:00   130.0      22.61  21.40   
12  CVX240315C00135000 2024-01-05 15:23:13+00:00   135.0      18.05  17.10   
13  CVX240315C00140000 2024-01-05 15:19:40+00:00   140.0      13.80  12.95   
14  CVX240315C00145000 2024-01-05 18:58:37+00:00   145.0       9.71   9.35   
15  CVX240315C00150000 2024-01-05 20:48:18+00:00   150.0       6.31   6.30   
16  CVX240315C00155000 2024-01-05 20:48:18+00:00   155.0       3.95   3.95   
17  CVX240315C00160000 2024-01-05 20:59:23+00:00   160.0       2.39   2.37   
18  CVX240315C00165000 2024-01-05 20:45:04+00:00   165.0       1.35   1.32   
19  CVX240315C00170000 2024-01-05 20:59:54+00:00   170.0       0.72   0.72   
20  CVX240315C00175000 2024-01-05 18:48:43+00:00   175.0       0.39   0.38   
21  CVX240315C00180000 2024-01-04 16:35:29+00:00   180.0       0.25   0.20   
22  CVX240315C00185000 2024-01-04 14:42:51+00:00   185.0       0.21   0.12   
23  CVX240315C00190000 2024-01-05 20:40:15+00:00   190.0       0.08   0.05   
24  CVX240315C00195000 2024-01-05 14:31:45+00:00   195.0       0.05   0.04   
25  CVX240315C00200000 2023-12-26 17:20:21+00:00   200.0       0.05   0.00   
26  CVX240315C00210000 2024-01-03 19:18:15+00:00   210.0       0.09   0.00   
27  CVX240315C00220000 2023-11-14 15:07:21+00:00   220.0       0.03   0.00   
28  CVX240315C00230000 2023-12-07 15:26:41+00:00   230.0       0.01   0.00   
29  CVX240315C00240000 2023-12-12 19:44:58+00:00   240.0       0.01   0.00   

      ask    change  percentChange  volume  openInterest  impliedVolatility  \
0   76.70  5.019997       7.102428     1.0             1           0.841798   
1   72.65  0.000000       0.000000    24.0            24           1.053716   
2   67.10  0.000000       0.000000     NaN             2           0.855226   
3   61.55  0.000000       0.000000     2.0            40           0.644047   
4   56.60  0.000000       0.000000     NaN             2           0.607914   
5   52.30  0.000000       0.000000     1.0            36           0.611088   
6   58.20  0.000000       0.000000     NaN             8           1.370609   
7   41.60  0.000000       0.000000     2.0            15           0.531255   
8   37.05  0.000000       0.000000     4.0            16           0.514897   
9   31.90  0.000000       0.000000     2.0           119           0.440191   
10  27.70  0.000000       0.000000     1.0            85           0.438482   
11  22.00  0.000000       0.000000     7.0           199           0.328620   
12  17.35 -0.400002      -2.168030     2.0           199           0.289924   
13  13.20  0.200000       1.470587    33.0           489           0.268806   
14   9.50 -0.410000      -4.051382    11.0          2687           0.251594   
15   6.40 -0.340000      -5.112784   222.0          1784           0.237923   
16   4.05 -0.400000      -9.195399   138.0          3496           0.229439   
17   2.43 -0.240000      -9.125476   160.0          6179           0.225166   
18   1.38 -0.150000      -9.999999   168.0          6916           0.222786   
19   0.78

In [12]:
cvx.option_chain(date).puts

contractSymbol             lastTradeDate  strike  lastPrice    bid  \
0   CVX240315P00075000 2023-12-06 15:46:27+00:00    75.0       0.04   0.00   
1   CVX240315P00080000 2023-12-11 20:18:41+00:00    80.0       0.02   0.00   
2   CVX240315P00085000 2023-12-11 20:18:01+00:00    85.0       0.05   0.00   
3   CVX240315P00090000 2023-11-29 15:07:37+00:00    90.0       0.06   0.00   
4   CVX240315P00095000 2023-11-17 20:52:56+00:00    95.0       0.15   0.02   
5   CVX240315P00100000 2024-01-04 20:13:42+00:00   100.0       0.05   0.02   
6   CVX240315P00105000 2023-12-26 15:01:51+00:00   105.0       0.08   0.03   
7   CVX240315P00110000 2024-01-03 20:32:23+00:00   110.0       0.11   0.10   
8   CVX240315P00115000 2024-01-04 16:45:21+00:00   115.0       0.18   0.17   
9   CVX240315P00120000 2024-01-05 16:06:39+00:00   120.0       0.28   0.29   
10  CVX240315P00125000 2024-01-05 20:45:47+00:00   125.0       0.50   0.49   
11  CVX240315P00130000 2024-01-05 16:06:39+00:00   130.0       0.88   0.85   
12  CVX240315P00135000 2024-01-05 19:38:09+00:00   135.0       1.48   1.47   
13  CVX240315P00140000 2024-01-05 20:47:28+00:00   140.0       2.49   2.45   
14  CVX240315P00145000 2024-01-05 19:47:28+00:00   145.0       4.00   3.90   
15  CVX240315P00150000 2024-01-05 20:51:45+00:00   150.0       6.00   5.95   
16  CVX240315P00155000 2024-01-05 18:08:00+00:00   155.0       8.45   8.65   
17  CVX240315P00160000 2024-01-04 15:14:14+00:00   160.0      10.65  12.05   
18  CVX240315P00165000 2024-01-05 17:09:24+00:00   165.0      16.23  16.15   
19  CVX240315P00170000 2024-01-05 19:55:53+00:00   170.0      21.08  20.05   
20  CVX240315P00175000 2023-12-06 19:53:37+00:00   175.0      31.85  24.50   
21  CVX240315P00180000 2023-11-07 17:42:29+00:00   180.0      37.90  36.00   
22  CVX240315P00185000 2023-09-11 16:42:38+00:00   185.0      21.67  25.35   
23  CVX240315P00190000 2023-09-22 17:34:23+00:00   190.0      24.91  25.45   
24  CVX240315P00200000 2023-09-06 17:47:50+00:00   200.0      33.55  37.70   
25  CVX240315P00230000 2023-10-27 19:27:37+00:00   230.0      86.42  83.65   

      ask    change  percentChange  volume  openInterest  impliedVolatility  \
0    0.08  0.000000       0.000000     5.0           610           0.613285   
1    0.08  0.000000       0.000000     8.0           362           0.560551   
2    0.08  0.000000       0.000000     7.0            36           0.511724   
3    0.09  0.000000       0.000000     1.0            93           0.509771   
4    0.11  0.000000       0.000000     2.0            70           0.473638   
5    0.11  0.020000      66.666670     5.0            49           0.425787   
6    0.14  0.000000       0.000000    10.0           188           0.394049   
7    0.14  0.000000       0.000000    10.0           325           0.349128   
8    0.21  0.000000       0.000000    14.0          1131           0.326179   
9    0.33 -0.010000      -3.448272     1.0          1166           0.306159   
10   0.54  0.050000      11.111114     7.0          2354           0.289314   
11   0.91  0.220000      33.333330     6.0          2348           0.275886   
12   1.54 -0.050000      -3.267971    17.0          3836           0.265266   
13   2.51 -0.020000      -0.796812   148.0          4685           0.254890   
14   4.00  0.150000       3.896107   423.0          5311           0.247566   
15   6.05  0.200000       3.448272   263.0          2792           0.239998   
16   8.80  0.210000       2.548544   207.0          1256           0.235481   
17  12.60  0.000000       0.000000     3.0          2578           0.250679   
18  16.45  1.250000       8.344460     6.0          1412           0.248909   
19  20.90  1.629999       8.380458     1.0           222           0.261482   
20  26.15  0.000000       0.000000   720.0           239           0.318122   
21  38.05  0.000000       0.000000     1.0             0           0.653080   
22  26.00  0.000000       0.000000     3.0             6           0.000010   
23  2

### Buying Calls to Bet on a Stock

- Investing in call options is similar to buying shares - you win when the stock goes up - but the % gains and losses are amplified
- It is similar to buying stocks with very high leverage
- The amount of "leverage" is greater for out of the money call
- Example: stock trading at 100.  Might buy calls with a strike of 110 or 120.
- Don't have to hold to maturity.  If the stock goes up, the price of the call will go up, and  you can sell and take the profit.

### Buying Puts to Bet Against a Stock

- Investing in puts is similar to shorting a stock - you win when the stock goes down - but the % gains and losses are amplified.
- The amplification is greater for out of the money puts.
- Example: stock trading at 100.  Might buy puts with a strike of 90 or 80.
- Again, don't have to hold to maturity.  

### Limited Liability

- Buying options is different in one way from buying stocks on margin or short selling - your maximum loss is capped with options.
- You can lose everything that you invest but not more than that.
- It is possible, though rare, to lose more than you've invested when buying a stock on margin or short selling.
- On the other hand, sellers of options have unlimited liability.

### Buying Puts for Protection

- If you own a stock, you might buy a put for insurance.
- You normally buy out-of-the-money puts for protection.
- Example: you own a stock trading at 100 and buy a put with a strike of 80.
- Your downside on the stock is now limited at 80, because you have an option to sell it at 80.
- Usually, you wouldn't exercise the put.  If the stock falls, you make a profit on the put that offsets some of your loss on the stock.


### Selling Calls for Income

- If you own a stock, you might sell a call option on it for income.  This is called selling a covered call.
- You normally sell out-of-the-money calls.
- Example: you own a stock trading at 100 and sell a call with a strike of 120.
- If the stock doesn't go up beyond 120, you'll be happy you sold the call.

### Collars

- It is common to buy a put for insurance and to sell a call to pay for the put.
- You are selling off some of your upside to protect your downside.
- Example: you own a stock trading at 100 and buy a put with a strike of 80 and sell a call with a strike of 120.
- The portfolio value is collared between 80 and 120.